service/frontend/workflowHandler.go

```diff
@@ -87,15 +87,16 @@ type (
 	WorkflowHandler struct {
 		resource.Resource
 
-		shuttingDown              int32
-		healthStatus              int32
-		tokenSerializer           common.TaskTokenSerializer
-		rateLimiter               quotas.Policy
-		config                    *Config
-		versionChecker            headers.VersionChecker
-		namespaceHandler          namespace.Handler
-		visibilityQueryValidator  *validator.VisibilityQueryValidator
-		searchAttributesValidator *validator.SearchAttributesValidator
+		shuttingDown                 int32
+		healthStatus                 int32
+		tokenSerializer              common.TaskTokenSerializer
+		rateLimiter                  quotas.Policy
+		config                       *Config
+		versionChecker               headers.VersionChecker
+		namespaceHandler             namespace.Handler
+		visibilityQueryValidator     *validator.VisibilityQueryValidator
+		searchAttributesValidator    *validator.SearchAttributesValidator
+		defaultWorkflowRetrySettings common.DefaultRetrySettings
```

Comment:
If we store this as a field on initialization of the handler, then it means we have to restart the server after updating the defaults in dynamic config.  It should be ok, but then I would explicitly call it out in a comment.
Another issue is it would prevent from converting this to a dynamic config with namespace filter.